packages/design-system/src/components/RichRadioButton/RichRadioButton.component.tsx

```diff
@@ -0,0 +1,69 @@
+// eslint-disable-next-line @talend/import-depth
+
+import { RichRadioButtonProps } from './RichRadioButton.types';
+import { StackVertical, StackHorizontal } from '../Stack';
+import { getTagVariant } from './RichRadioButton.utils';
+import { getIconWithDeprecatedSupport } from '../Icon/DeprecatedIconHelper';
+import style from './RichRadioButton.module.scss';
+
+const RichRadioButton = ({
+	dataFeature,
+	description,
+	asset,
+	id,
+	isChecked = false,
+	isDisabled = false,
+	isReadOnly = false,
+	name,
+	onChange,
+	tags,
+	title,
+}: RichRadioButtonProps) => {
+	return (
+		<label className={style['rich-radio-button__wrapper']}>
+			<input
+				type="radio"
+				id={id}
+				name={name}
+				disabled={isDisabled}
+				readOnly={isReadOnly}
+				data-feature={dataFeature}
+				checked={isChecked}
+				onClick={() => onChange(id)}
+			/>
+			<span className={style['rich-radio-button']}>
+				<StackVertical as="span" gap="XS">
+					{asset && (
+						<>
+							<span
+								className={
+									style[`rich-radio-button__${asset.illustration ? 'illustration' : 'icon'}`]
```

Comment:
maybe it could be more readable using `classnames`, with for example something like : 

```js
classnames({
    'rich-radio-button__illustration': asset.illustration,
    'rich-radio-button__icon': !asset.illustration,
})
```